### PR TITLE
Seeded PRNG utility and run configuration

### LINKED ISSUE #12: Seeded PRNG utility and run configuration
## Objective
Implement a deterministic random number generator seeded by a per-run value, ensuring that the same seed always produces the same procedural world. This is the foundation for reproducible runs and shareable seeds.

## Acceptance Criteria
- [ ] `src/lib/rng.ts` wraps the seedrandom package with a typed API exposing common random operations (float, int range, pick from array, weighted pick, shuffle)
- [ ] `RunConfig` type defined holding seed (string), and other per-run parameters
- [ ] Determinism verified: calling the same sequence of random operations with the same seed produces identical results across runs
- [ ] Seed generation utility for new runs (e.g., random alphanumeric string)
- [ ] RNG instance is injectable/resettable for testing purposes
- [ ] All procedural systems will consume this shared RNG — no use of Math.random() in game code

## Technical Notes
- Use the `seedrandom` npm package for the underlying PRNG algorithm
- The RNG wrapper should be a class or factory that produces a fresh instance per run
- Seed display on the death screen (implemented in M5) will use the seed from RunConfig
- Consider providing convenience methods: `rng.float()`, `rng.intBetween(min, max)`, `rng.pick(array)`, `rng.weightedPick(items, weights)`, `rng.shuffle(array)`

## Dependencies
- Scaffold Next.js 16 project with game dependency set

### PR DESCRIPTION
**Pull request by allan-mobley-jr**
## Summary
Closes #12: Adds a deterministic seeded PRNG utility wrapping seedrandom and a RunConfig type for per-run session parameters.

## Changes
- `src/lib/rng.ts` — RNG class with float, int, pick, weightedPick, shuffle, derive, reset, raw methods + generateSeed utility
- `src/types/run.ts` — RunConfig type (seed, difficulty, targetMinutes) with RUN_DEFAULTS
- `src/lib/rng.test.ts` — 63 test cases covering all methods, edge cases, determinism, and error paths
- `src/types/run.test.ts` — RUN_DEFAULTS value regression guard
- **Bug fix**: `int()` collapsed-range branch now advances PRNG state to maintain deterministic sequence position

## Test Coverage
- 63 tests in rng.test.ts (59 from Blacksmith + 4 new: collapsed-range state advancement, stale raw() after reset, derive() parent isolation, RUN_DEFAULTS values)
- 1 test in run.test.ts (RUN_DEFAULTS regression guard)
- 1 bug found and fixed during testing: `int()` with fractional bounds that collapse after rounding (e.g. `int(1.5, 1.9)`) skipped the PRNG call, causing sequence position to depend on argument values

## Acceptance Criteria
- [x] `src/lib/rng.ts` wraps seedrandom with typed API (float, int range, pick, weighted pick, shuffle)
- [x] `RunConfig` type defined with seed, difficulty, targetMinutes
- [x] Determinism verified: same seed + same operations = identical results
- [x] Seed generation utility (`generateSeed`) using crypto.getRandomValues
- [x] RNG injectable (`new RNG(seed)`) and resettable (`reset()`) for testing
- [x] No use of Math.random() in game code — shared RNG infrastructure ready

## Quality Checks
- [x] Lint passes
- [x] Type check passes
- [x] Tests pass (96 total)
- [x] Build succeeds

### CODE REVIEW

File: src/lib/rng.ts [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+   *
+   * @param namespace - A unique label for the subsystem (e.g. `"wfc"`, `"bsp"`, `"loot"`).
+   * @throws {RangeError} If `namespace` is empty.
+   */
+  derive(namespace: string): RNG {
+    if (namespace.length === 0) {
+      throw new RangeError('RNG.derive: namespace must not be empty');
+    }
+    return new RNG(`${this._seed}-${namespace}`);
+  }
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply src/lib/rng.ts]: This is actually handled by the architecture — seeds are generated by `generateSeed()` which produces alphanumeric-only strings (A-Z, a-z, 0-9), never containing `-`. Namespaces are hardcoded labels like `"wfc"`, `"bsp"`, `"loot"`. The collision scenario (seed `"a-b"` + ns `"c"` vs seed `"a"` + ns `"b-c"`) requires the seed to contain `-`, which `generateSeed()` cannot produce.

File: src/lib/rng.test.ts [hunk 2]
```diff
@@ -0,0 +1,602 @@
...
+  });
+
+  it('accepts an empty string seed without throwing', () => {
+    expect(() => new RNG('')).not.toThrow();
+  });
+
+  it('accepts unicode seeds', () => {
+    const rng = new RNG('');
+    expect(rng.seed).toBe('');
+    expect(rng.float()).toBeGreaterThanOrEqual(0);
```
[the remark this answers is not in the document]
Comment (allan-mobley-jr) [reply src/lib/rng.test.ts]: The test already uses a unicode seed — `new RNG('🎮')` (gamepad emoji U+1F3AE). The emoji may not render in the diff view, but it is present in the source. The test verifies the seed is stored correctly and produces a valid float, which is sufficient for confirming unicode handling.
